module4/auth/src/main/java/com/epam/esm/security/filter/JwtTokenProvider.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.esm.security.filter;
+
+import com.epam.esm.security.exception.InvalidTokenException;
+import com.epam.esm.security.exception.TokenExpiredException;
+import com.epam.esm.service.UserService;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.SignatureException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.naming.AuthenticationException;
+import javax.servlet.http.HttpServletRequest;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Date;
+
+import static org.springframework.util.StringUtils.hasText;
+
+@Component
+@RequiredArgsConstructor
+public class JwtTokenProvider {
+
+    private final UserService userService;
+
+    @Value("${jwt.secret}")
+    private String secretKey;
+    @Value("${jwt.expiration}")
+    private long validityInSeconds;
+    @Value("${jwt.header}")
+    private String authorizationHeader;
+
+    @PostConstruct
+    protected void init() {
+        secretKey = Base64.getEncoder().encodeToString(secretKey.getBytes());
+    }
+
+    public String createToken(String username) {
+        Instant instant = Instant.now();
+        Instant expiration = instant.plus(validityInSeconds, ChronoUnit.MINUTES);
+
+        return Jwts.builder()
+                .setSubject(username)
+                .setIssuedAt(Date.from(instant))
+                .setExpiration(Date.from(expiration))
+                .signWith(SignatureAlgorithm.HS256, secretKey)
+                .compact();
+    }
+
+    public boolean validateToken(String token) throws AuthenticationException {
+        try {
+            Jws<Claims> claimsJws = Jwts.parser()
+                    .setSigningKey(secretKey)
+                    .parseClaimsJws(token);
+            return !claimsJws.getBody().getExpiration().before(new java.util.Date());
```

Comment:
why with package java.util.Date() and why not Date.from(instant)?

Reply:
used Date.from(instant)